### PR TITLE
chore: Moves Card to the components folder

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Moves `Card` to the components folder.

This work is part of [SIP-61](https://github.com/apache/superset/issues/13632)

@rusackas @junlincc

### TEST PLAN
1 - Execute all tests
2 - All tests should pass

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by pkdotson — APPROVED** [review]
LGTM!

File: superset-frontend/src/components/Card/index.tsx [hunk 1]
```diff
@@ -17,23 +17,24 @@
  * under the License.
  */
 import React from 'react';
-import { styled } from '@superset-ui/core';
+import { SupersetTheme } from '@superset-ui/core';
```
Comment (rusackas) [inline superset-frontend/src/components/Card/index.tsx:20]: I don't like using the SupersetTheme directly like this... we can `useTheme` if you're set on the `css` method, but I want to maintain support for the `theme` provided by the `ThemeProvider` which effectively mutates/overrides the `SupersetTheme`

**Review by rusackas — CHANGES_REQUESTED** [review]
Trying to avoid using `SupersetTheme` directly, but the rest of it LGTM.

File: superset-frontend/src/components/Card/index.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/Card/index.tsx:20]: Here we are not using the theme object. It's just the theme type. The values are coming from the `ThemeProvider`.
`css={(theme: SupersetTheme) => ({..})}`

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**